api/src/legacy_soap_api/legacy_soap_api_auth.py

```diff
@@ -100,3 +113,45 @@ def get_soap_client_certificate(urlencoded_cert: str) -> SOAPClientCertificate:
         issuer=cert.issuer.rfc4514_string(),
         serial_number=cert.serial_number,
     )
+
+
+def validate_certificate(
+    db_session: db.Session, soap_auth: SOAPAuth | None, api_name: str
+) -> LegacyCertificate:
+    if not soap_auth:
+        logger.warning(
+            "soap_client_certificate: no soap auth",
+        )
+        raise SOAPClientCertificateLookupError("no soap auth")
+
+    serial_number_str = str(soap_auth.certificate.serial_number)
+    legacy_certificate = (
+        db_session.query(LegacyCertificate)
+        .filter(LegacyCertificate.serial_number == serial_number_str)
+        .first()
+    )
+    extra_data = {"serial_number": serial_number_str}
+
+    if not legacy_certificate:
+        logger.warning(
+            "soap_client_certificate: could not retrieve client cert for serial number",
+            extra=extra_data,
+        )
+        raise SOAPClientCertificateLookupError("could not retrieve client cert for serial number")
+
+    extra_data.update({"legacy_certificate_id": str(legacy_certificate.legacy_certificate_id)})
```

Comment:
Slightly shorter way to do this - don't need to stringify UUIDs, our logging automatically converts most types in the extra - can see how those work in https://github.com/HHS/simpler-grants-gov/blob/main/api/src/logging/formatters.py#L29

Can also do `|=` instead of update to trim a little out.

```suggestion
    extra_data |= {"legacy_certificate_id":legacy_certificate.legacy_certificate_id}
```